tensorboard/webapp/widgets/linked_time_fob/linked_time_fob_controller_component.ts

```diff
@@ -0,0 +1,190 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import {LinkedTime} from '../../metrics/types';
+import {ScaleLinear, ScaleTime} from '../../third_party/d3';
+
+export enum AxisDirection {
+  horizontal = 0,
+  vertical,
+}
+
+export enum DraggingFob {
```

Comment:
The "Dragging" wording was confusing to me. I think this is used to distinguish which fob the user is interacting with. Could we get ride of the Dragging part and just `Fobs`?

Reply:
then we can also drop Fob inside
```
enum Fobs {
NONE,
START,
END,
}
```